### PR TITLE
Create fibonacci_formula.c

### PR DESCRIPTION
**Pull request by GrandSir**
#### Description of Change
create fibonacci algorithm using explicit formula

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by Chzyi** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

File: misc/fibonacci_formula.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * explicit formula solution](https://fabiandablander.com/r/Fibonacci.html)
+ *
+*/
+
+#include <math.h> /// for pow and sqrt
+#include <stdio.h>
+
+/*
+ * formula comes from quadratic formula and recurrence relation
+ * all the proof can be found in the related article in top of the file
```
Comment (Panquesito7) [inline misc/fibonacci_formula.c:14]: Missing parameter documentation.


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not maintainable

File: misc/fibonacci_formula.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/* print first 10 elements of fibonacci sequence for testing */
+static void test () {
+    for(int i = 0; i <= 10; i++){
+        printf("%d. fibonacci sequence is: %.0f\n", i, fib(i));
+    }
+}
+
+// driver code
+int main() {
+    test();
```
Comment (Panquesito7) [inline misc/fibonacci_formula.c:30]:
```suggestion
    test();  // run self-test implementations
```


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Tags:** Inelegant

File: misc/fibonacci_formula.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+/* print first 10 elements of fibonacci sequence for testing */
+static void test () {
+    for(int i = 0; i <= 10; i++){
+        printf("%d. fibonacci sequence is: %.0f\n", i, fib(i));
+    }
+}
+
+// driver code
```
Comment (Panquesito7) [inline misc/fibonacci_formula.c:28]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
```


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not maintainable

File: misc/fibonacci_formula.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+/*
+ * formula comes from quadratic formula and recurrence relation
+ * all the proof can be found in the related article in top of the file
+ */
+double fib(int n) {
+    return (1 / sqrt(5) * pow(((1 + sqrt(5)) / 2), n + 1)) -
+           (1 / sqrt(5) * pow(((1 - sqrt(5)) / 2), n + 1));
+}
+
+/* print first 10 elements of fibonacci sequence for testing */
```
Comment (Panquesito7) [inline misc/fibonacci_formula.c:21]:
```suggestion
/**
 * @brief Self-test implementations
 * @returns void
 */
```


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not maintainable

File: misc/fibonacci_formula.c [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+static void test () {
+    for(int i = 0; i <= 10; i++){
+        printf("%d. fibonacci sequence is: %.0f\n", i, fib(i));
```
Comment (Panquesito7) [inline misc/fibonacci_formula.c:24]: Please add self-test cases using `assert` (included in the `assert.h` library/header) so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or on our [**Discord server**](https://discord.gg/c7MnfGFGa6). 😃


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inefficient

File: misc/fibonacci_formula.c [hunk 6]
```diff
@@ -0,0 +1,32 @@
+/**
+ * \file
+ *
+ * \brief Implementation of [Fibonacci Sequence
+ * explicit formula solution](https://fabiandablander.com/r/Fibonacci.html)
```
Comment (Panquesito7) [inline misc/fibonacci_formula.c:5]: Would you like to provide a detailed description of the algorithm? You can include references from Wikipedia and other sources as long as it's not plagiarism. 🙂


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question&nbsp; | &nbsp;**Tags:** Inelegant

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! 🎉

File: misc/fibonacci_formula.c (hunk 6 above)
Comment (GrandSir) [reply misc/fibonacci_formula.c:5]: Actually, it is only based on fibonacci formula which is derived from a bunch of another formulas. So the proof is not algorithmic but mathematical. but I can surely prove it in comment if requested.

Comment (GrandSir) [reply misc/fibonacci_formula.c:5]: I searched wikipedia for the proof of the algorithm but it is far more complex and none of them are using explicit formula

File: misc/fibonacci_formula.c [hunk 7]
```diff
@@ -0,0 +1,58 @@
+    /**
+     * @file
+     * @brief [finding Fibonacci number of any number n using binet's closed form formula](https://en.wikipedia.org/wiki/Fibonacci_number#Binet's_formula)
+     * @details
+     * compute \f$f_{nth}\f$ Fibonacci number using the binet's formula:
+     * Fn = 1√5 * (1+√5 / 2)^n+1 − 1√5 * (1−√5 / 2)^n+1
+     * @author [GrandSir](https://github.com/GrandSir/)
+    */
```
Comment (Panquesito7) [inline misc/fibonacci_formula.c:8]: Seems there's extra indentation here.
The description looks good, by the way.
```suggestion
/**
 * @file
 * @brief [finding Fibonacci number of any number n using binet's closed form formula](https://en.wikipedia.org/wiki/Fibonacci_number#Binet's_formula)
 * compute \f$f_{nth}\f$ Fibonacci number using the binet's formula:
 * Fn = 1√5 * (1+√5 / 2)^n+1 − 1√5 * (1−√5 / 2)^n+1
 * @author [GrandSir](https://github.com/GrandSir/)
 */
```


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inelegant

**Review by Panquesito7 — COMMENTED** [review]
Almost there! 😄
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/GrandSir/C/actions


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by GrandSir** [comment]
all done, hope there's nothing else.

File: misc/fibonacci_formula.c [hunk 8]
```diff
@@ -0,0 +1,57 @@
+/**
+ * @file
+ * @brief [finding Fibonacci number of any number n using binet's closed form formula](https://en.wikipedia.org/wiki/Fibonacci_number#Binet's_formula)
```
Comment (Panquesito7) [inline misc/fibonacci_formula.c:3]:
```suggestion
 * @brief Finding Fibonacci number of any `n` number using [Binet's closed form formula](https://en.wikipedia.org/wiki/Fibonacci_number#Binet's_formula)
```


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Tags:** Inelegant

File: misc/fibonacci_formula.c [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    assert(fib(1) == 1);
+    assert(fib(2) == 2);
+    assert(fib(3) == 3);
+    assert(fib(4) == 5);
+    assert(fib(5) == 8);
+    assert(fib(6) == 13);
+    assert(fib(7) == 21);
+    assert(fib(8) == 34);
+    assert(fib(9) == 55);
+    assert(fib(10) == 89);
```
Comment (Panquesito7) [inline misc/fibonacci_formula.c:43]:
```suggestion
    assert(fib(10) == 89);
    
    printf("All tests have successfully passed!\n");
```


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inelegant

**Review by Panquesito7 — COMMENTED** [review]
Almost there! 🎉

File: misc/fibonacci_formula.c (hunk 9 above)
Comment (GrandSir) [reply misc/fibonacci_formula.c:43]: ahh... really?


why this is required though?

Comment (GrandSir) [reply misc/fibonacci_formula.c:43]: okay anyways, haha.

**Review by Panquesito7 — APPROVED** [review]
Amazing work! Thanks so much for contributing. 🚀

**Review by mishraabhinn — APPROVED** [review]

**Pull request merged**